.github/workflows/ci.yml

```diff
@@ -373,6 +373,138 @@ jobs:
         run: env
         if: ${{ always() }}
 
+  sanitizers_debian:
+    name: ${{ matrix.configuration.job_name }}
+    runs-on: ubuntu-latest
+    needs: docker_cache
+
+    strategy:
+      fail-fast: false
+      matrix:
+        configuration:
+          - job_name: 'Valgrind (memcheck)'
+            env_vars:
+              # The `--error-exitcode` is required to make the test fail if valgrind found errors,
+              # otherwise it will return 0 (https://www.valgrind.org/docs/manual/manual-core.html).
+              WRAPPER_CMD: 'valgrind --error-exitcode=42'
+              SECP256K1_TEST_ITERS: 2
+          - job_name: 'UBSan, ASan, LSan'
+            env_vars:
+              CFLAGS: '-fsanitize=undefined,address -g'
+              UBSAN_OPTIONS: 'print_stacktrace=1:halt_on_error=1'
+              ASAN_OPTIONS: 'strict_string_checks=1:detect_stack_use_after_return=1:detect_leaks=1'
+              LSAN_OPTIONS: 'use_unaligned=1'
+              SECP256K1_TEST_ITERS: 32
+        asm:
+          - env_vars: { ASM: 'auto' }
+          - env_vars: { ASM: 'no', ECMULTGENPRECISION: 2, ECMULTWINDOW: 2 }
+        cc:
+          - env_vars: { CC: 'clang' }
+          - env_vars: { HOST: 'i686-linux-gnu', CC: 'i686-linux-gnu-gcc' }
+
+    env:
+      ECDH: 'yes'
+      RECOVERY: 'yes'
+      SCHNORRSIG: 'yes'
+      ELLSWIFT: 'yes'
+      CTIMETESTS: 'no'
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set environment variables
+        run: |
+          for var in "$(jq -s 'add' <<< '${{ toJSON(matrix.configuration.env_vars) }} ${{ toJSON(matrix.asm.env_vars) }} ${{ toJSON(matrix.cc.env_vars) }}' | jq -r 'to_entries | .[] | "\(.key)=\(.value)"')"; do
+            echo "$var" >> "$GITHUB_ENV"
+          done          
```

Comment:
A bit ugly, hm... There's no better way to merge this, I guess.

I think it makes sense to split this job into two jobs "Valgrind (memcheck)" and "UBSan, ASan, LSan". Then for each of these jobs, we can expand the remaining 2x2 combinations, i.e., list the 4 resulting combinations explicitly. This is probably a good idea anyway, it will also make it a bit easier to add variety to the jobs.

What do you think?

Reply:
> I think it makes sense to split this job into two jobs "Valgrind (memcheck)" and "UBSan, ASan, LSan".

I was thinking about the same. But it won't help.

Only one `env` property is allowed job-wide and step-wide. Job-wide `env` is already busy with global variables. Therefore, even 2 sets won't fit one step-wide `env`.

Reply:
> list the 4 resulting combinations explicitly

... but that is completely feasible.

Reply:
> > list the 4 resulting combinations explicitly
> 
> ... but that is completely feasible.

Done.